src/DynamoCore/Utilities/NotificationCounter.cs

```diff
@@ -0,0 +1,213 @@
+    {
+        internal SortedSet<int> objects = new SortedSet<int>();
+        internal int counter;
+    }
+
+    internal static class NotificationCounter
+    {
+        private static Harmony harmonyIntance;
+        private static bool patchedCoreNotifications;
+        private static bool patchUINotifications;
+
+        private static double timerInterval = 5000;//5 second;
+        private static Timer timer;
+        private static string logFile = "";
+        private static Dictionary<string, NotificationLog> notifications = new Dictionary<string, NotificationLog>();
+
+        internal static void StartCounting()
+        {
+            try
+            {
+                lock (logFile)
+                {
+                    logFile = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "notification_counter.log");
+                    if (File.Exists(logFile))
+                    {
+                        File.Delete(logFile);
+                    }
+                }
+
+                timer = new Timer();
+                timer.Elapsed += Timer_Elapsed;
+                timer.Interval = timerInterval;
+                timer.Start();
+                
+                harmonyIntance = new Harmony("NotificationCounter");
+
+                AppDomain.CurrentDomain.AssemblyLoad += new AssemblyLoadEventHandler((object sender, AssemblyLoadEventArgs args) => PatchFromAssmebly(args.LoadedAssembly));
+
+                var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies().ToList();
+                PatchFromAssmebly(loadedAssemblies.FirstOrDefault(x => x.GetName().Name == "DynamoCore"));
+                PatchFromAssmebly(loadedAssemblies.FirstOrDefault(x => x.GetName().Name == "Microsoft.Practices.Prism"));
+                PatchObservableCollection();
+            }
+            catch { }
+        }
+
+        private static void Timer_Elapsed(object sender, ElapsedEventArgs e)
+        {
+            try
+            {
+                if (notifications.Count == 0) return;
+
+                bool freshLog = !File.Exists(logFile);
+                Dictionary<string, (int counter, int objects)> localNotifications = new Dictionary<string, (int counter, int objects)>();
+                
+                lock (notifications)
+                {
+                    foreach (var item in notifications)
+                        localNotifications.Add(item.Key, (item.Value.counter, item.Value.objects.Count));
+
+                    notifications.Clear();
+                }
+
+                lock (logFile)
+                {
+                    if (File.Exists(logFile))
+                    {
+                        var oldContent = File.ReadAllLines(logFile);
+                        mergeLogContents(localNotifications, oldContent);
+                    }
+                }
+
+                var local2 = localNotifications.Select(x => Tuple.Create(x.Key, x.Value.Item1, x.Value.Item2)).ToList();
```

Comment:
I know this is just developer code, but named tuples are quite nice for readability.